test/pytest/test_auto_precision.py

```diff
@@ -0,0 +1,255 @@
+    )
+    model.add(AveragePooling2D(pool_size=(2, 2), name='first_pool'))
+    model.add(ReLU(name='first_act'))
+    model.add(Conv2D(4, kernel_size=(3, 3), activation='relu', name='middle_layer'))
+    model.add(Conv2D(4, kernel_size=(1, 1), activation='relu', name='last_layer'))  # Will become PointwiseConv2D
+    model.add(Flatten())
+    model.add(Dense(4, activation='relu'))
+    model.compile()
+    return model
+
+
+@pytest.fixture(scope='module')
+def keras_model_sepconv1d():
+    model = Sequential()
+    model.add(SeparableConv1D(8, kernel_size=3, activation='linear', name='first_layer', input_shape=(in_width, in_feat)))
+    model.add(AveragePooling1D(pool_size=2, name='first_pool'))
+    model.add(ReLU(name='first_act'))
+    model.add(Conv1D(4, kernel_size=2, activation='relu', name='middle_layer'))
+    model.add(Conv1D(4, kernel_size=1, activation='relu', name='last_layer'))  # Will become PointwiseConv1D
+    model.add(Flatten())
+    model.add(Dense(4, activation='relu'))
+    model.compile()
+    return model
+
+
+@pytest.fixture(scope='module')
+def keras_model_sepconv2d():
+    model = Sequential()
+    model.add(
+        SeparableConv2D(
+            8, kernel_size=(3, 3), activation='linear', name='first_layer', input_shape=(in_height, in_width, in_feat)
+        )
+    )
+    model.add(AveragePooling2D(pool_size=(2, 2), name='first_pool'))
+    model.add(ReLU(name='first_act'))
+    model.add(Conv2D(4, kernel_size=(3, 3), activation='relu', name='middle_layer'))
+    model.add(Conv2D(4, kernel_size=(1, 1), activation='relu', name='last_layer'))  # Will become PointwiseConv2D
+    model.add(Flatten())
+    model.add(Dense(4, activation='relu'))
+    model.compile()
+    return model
+
+
+@pytest.mark.parametrize('io_type', ['io_stream', 'io_parallel'])
+@pytest.mark.parametrize('backend', ['Vivado', 'Vitis', 'Quartus'])
+@pytest.mark.parametrize('model_type', ['conv1d', 'conv2d'])
+def test_auto_precision_conv(keras_model_conv1d, keras_model_conv2d, data_2d, data_3d, model_type, io_type, backend):
+    if model_type == 'conv1d':
+        model = keras_model_conv1d
+        data = data_2d
+    else:
+        model = keras_model_conv2d
+        data = data_3d
+
+    config = hls4ml.utils.config_from_keras_model(model, default_precision='ap_fixed<16,6>', granularity='model')
+    config['LayerName'] = {
+        # Infer all types of these layers
+        'first_layer': {
+            'Precision': 'auto',
+        },
+        'first_pool': {
+            'Precision': 'auto',
+        },
+        # Infer only a few specific types for these layers
+        'middle_layer': {
+            'Precision': {
+                'accum': 'auto',
+                'weight': 'auto',
+            },
+        },
+        'last_layer': {
+            'Precision': {
+                'result': 'auto',
+            },
+        },
+    }
+    odir = str(test_root_path / f'hls4mlprj_auto_{model_type}_{backend}_{io_type}')
```

Comment:
Might want to add something like this to avoid Winograd, which is lower precision:
```python
    if backend == 'Quartus' and io_type == 'io_parallel':
        # Winegrad imp[lementation does not support binary
        hls_config["LayerName"]["conv2d_1"]["Implementation"] = "im2col"
        hls_config["LayerName"]["conv2d_2"]["Implementation"] = "im2col"
        hls_config["LayerName"]["conv2d_3"]["Implementation"] = "im2col"
```